src/test/java/us/kbase/test/narrativemethodstore/db/mongo/MongoDBHelper.java

```diff
@@ -51,45 +52,10 @@ private static int startupMongo(String mongodExePath, File dir) throws Exception
             mongodExePath = "mongod";
         if (!dir.exists())
             dir.mkdirs();
-        File dataDir = new File(dir, "data");
-        dataDir.mkdir();
-        File logFile = new File(dir, "mongodb.log");
-        int port = findFreePort();
-        File configFile = new File(dir, "mongod.conf");
-        TextUtils.writeLines(Arrays.asList(
-                "dbpath=" + dataDir.getAbsolutePath(),
-                "logpath=" + logFile.getAbsolutePath(),
-                "logappend=true",
-                "port=" + port,
-                "bind_ip=127.0.0.1"
-                ), configFile);
-        File scriptFile = new File(dir, "start_mongo.sh");
-        TextUtils.writeLines(Arrays.asList(
-                "#!/bin/bash",
-                "cd " + dir.getAbsolutePath(),
-                mongodExePath + " --nojournal --config " + configFile.getAbsolutePath() + " >out.txt 2>err.txt & pid=$!",
-                "echo $pid > pid.txt"
-                ), scriptFile);
-        ProcessHelper.cmd("bash", scriptFile.getCanonicalPath()).exec(dir);
-        boolean ready = false;
-        int waitSec = 120;
-        for (int n = 0; n < waitSec; n++) {
-            Thread.sleep(1000);
-            if (logFile.exists()) {
-                if (TextUtils.grep(TextUtils.lines(logFile), "waiting for connections on port " + port).size() > 0) {
-                    ready = true;
-                    break;
-                }
-            }
-        }
-        if (!ready) {
-            if (logFile.exists())
-                for (String l : TextUtils.lines(logFile))
-                    System.err.println("MongoDB log: " + l);
-            throw new IllegalStateException("MongoDB couldn't startup in " + waitSec + " seconds");
-        }
-        System.out.println(dir.getName() + " was started up");
-        return port;
+        MongoController mongo = new MongoController(mongodExePath, dir.toPath(), true);
+        System.out.println(String.format("Testing against mongo executable %s on port %s",
+                mongodExePath, mongo.getServerPort()));
+        return mongo.getServerPort();
```

Comment:
Wow that's some old code. That being said, keep a reference to `mongo` and use it's destroy() method rather than the current pid killing approach

Reply:
👍

Reply:
:+1: